### PR TITLE
Update Device Health fields description for KEP-4680

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#### What type of PR is this?

/kind cleanup
/kind documentation

KEP: https://github.com/kubernetes/enhancements/issues/4680

Just aligning with the latest KEP update.

/sig node

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
@thockin can you please review this? This is mostly comments, and they were reviewed during the KEP review already.

Also it is a small pre-work for the DRA support.

/assign @thockin

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -2788,11 +2788,17 @@ type ContainerStatus struct {
 }
 
 type ResourceStatus struct {
+	// Name of the resource. Must be unique within the pod and in case of non-DRA resource, match one of the resources from the pod spec.
+	// For DRA resources, the value must be claim:claim_name/request.
```
Comment (thockin) [inline pkg/apis/core/types.go:2792]: `"claim:<claim_name>/<request>"` ?

File: pkg/apis/core/types.go [hunk 2]
```diff
@@ -2788,11 +2788,17 @@ type ContainerStatus struct {
...
+	// The claim_name must match one of the claims from resourceClaims field in the podSpec.
+	// +required
 	Name ResourceName
-	// List of unique Resources health. Each element in the list contains a unique resource ID and resource health.
+	// List of unique Resources health. Each element in the list contains an unique resource ID and resource health.
```
Comment (thockin) [inline pkg/apis/core/types.go:2796]: don't capitalize Resource unless it's another type, but if so, capitalize above?

File: pkg/apis/core/types.go [hunk 3]
```diff
@@ -2801,12 +2807,12 @@ type ResourceStatus struct {
 // ResourceID is calculated based on the source of this resource health information.
 // For DevicePlugin:
 //
-//	deviceplugin:DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
+//	DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
```
Comment (thockin) [inline pkg/apis/core/types.go:2810]: If this is a public API doc, can you write it in English without a reference to code?

File: pkg/apis/core/validation/validation.go [hunk 4]
```diff
@@ -8260,17 +8260,39 @@ func validateContainerStatusAllocatedResourcesStatus(containerStatuses []core.Co
...
+						if name == string(allocatedResource.Name) {
+							found = true
+							break
+						}
+					}
+
+				} else {
+					// assume it is a resource name
+
+					errorStr = "must match one of the container's resource requirements"
```
Comment (thockin) [inline pkg/apis/core/validation/validation.go:8285]: s/requirements/requests? I am not sure on this one.

File: pkg/apis/core/types.go (hunk 1 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go]: fixed

File: pkg/apis/core/types.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go]: improved wording

File: pkg/apis/core/types.go (hunk 3 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/types.go]: yes, fixed and also referfenced the pod resource API which is how end user can see it

File: pkg/apis/core/validation/validation.go (hunk 4 above)
Comment (SergeyKanzhelev) [reply pkg/apis/core/validation/validation.go]: thanks!

**Comment by SergeyKanzhelev** [comment]
@thockin thank you for your review!

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**